src/evergreen.ts

```diff
@@ -188,21 +188,32 @@ export class client {
         this.postAPIResource(callback, apiV2Resource("admin/settings"), body);
     }
 
-    private formRequest(url: string, body?: any): requestOpts {
+    private formRequest(url: string, requireHeaders: boolean, body?: any): requestOpts {
```

Comment:
There's more duplicated code than there needs to be here. I think the only thing that needs to change is to add the headers if requireHeaders is sent, otherwise everything can stay the same

Reply:
Fixed